config/hostcredentials/transform.go

```diff
@@ -0,0 +1,80 @@
+// Package hostcredentials implements a TransformationProvider that fetches
+// credentials from the (oddly named) `host-secrets` service and replaces
+// objects of the form: {$hostcredentials: [url, url]} with the credentials.
+//
+// The given URLs should point to the `/v1/credentials` endpoint of instances of the
+// [taskcluster-host-secrets](https://github.com/taskcluster/taskcluster-host-secrets)
+// service.  They will be tried in order until success.  This is a simple form
+// of client-side resilience to failure of a single instance.
+//
+// Note that this transform will need to run before the "hostsecrets" transform.
+package hostcredentials
+
+import (
+	"encoding/json"
+	got "github.com/taskcluster/go-got"
+	"log"
+	"time"
+
+	"github.com/taskcluster/taskcluster-worker/config"
+)
+
+type provider struct{}
+
+func init() {
+	config.Register("hostcredentials", provider{})
+}
+
+func (provider) Transform(cfg map[string]interface{}) error {
+	g := got.New()
+
+	return config.ReplaceObjects(cfg, "hostcredentials", func(val map[string]interface{}) (interface{}, error) {
+		var urls []string
+		for _, u := range val["$hostcredentials"].([]interface{}) {
+			urls = append(urls, u.(string))
+		}
+
+		var creds struct {
+			ClientId    string `json:"clientId"`
+			AccessToken string `json:"accessToken"`
+			Certificate string `json:"certificate"`
+		}
+
+	FetchLoop:
+		for {
+			for _, url := range urls {
+				log.Printf("Trying host-secrets server %s...", url)
+
+				resp, err := g.Get(url).Send()
+				if err != nil {
+					log.Printf("result: %s; continuing to next server", err)
+					continue
+				}
+
+				if resp.StatusCode != 200 {
+					log.Printf("result: HTTP %d; continuing to next server", resp.StatusCode)
+					continue
+				}
+
+				err = json.Unmarshal(resp.Body, &creds)
+				if err != nil {
+					log.Printf("decoding JSON from server: %s; continuing to next server", err)
+					continue
+				}
+
+				break FetchLoop
```

Comment:
I would just do a `return creds` here... maybe defined `type credentials struct {...} ` at top-level...

But strive to avoid the nested loop breaking magic... In fact I would probably just create a `fetchCredentials(urls []string) (credentials, error)` function...

Btw, are you sure you wish to loop forever? it's okay with me, I just want to be sure... you can return an error which will be printed and the worker will crash whatever that means.